Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,300 @@
+# Third Party Dependencies
+
+This document aims to describe the current issues surrounding the use of 
+third-party libraries within the mantid codebase,
+along with proposals for solutions to these problems.
+
+## Motivation
+
+Mantid uses a wide variety of third-party libraries
+and takes different approaches for managing them on each operating system - 
+this vastly increases the management overhead for any library used. 
+Furthermore each OS uses a different version of given package
+making it impossible to define a single version across the whole project.
+
+The motivation behind this work is to vastly reduce the dependency management
+overhead across the project.
+
+## Current Situation
+
+The following sections describe the current state of play for 
+3rd-party library management for each of our supported operating systems.
+
+### Linux (RHEL 7 & Ubuntu)
+
+Linux-based platforms use OS-provided system packages from distro repositories
+or custom builds of libraries where distro-provided versions are
+too old.
+See 
+[copr for RPM](https://copr.fedorainfracloud.org/coprs/mantid/mantid/)
+and 
+[launchpad PPA for Ubuntu](https://launchpad.net/~mantid/+archive/ubuntu/mantid).
+[Developer packages](https://github.com/mantidproject/mantid/tree/master/buildconfig/dev-packages)
+are provided for developers to easily install the required dependencies.
+
+Upgrading libraries is possible by placing newer versions in the above locations.
+For some libraries, such as HDF5 or sip, 
+this is not advised due to conflicts with other applications requiring
+different versions.
+A real example is with the rebuilt `sip` package for RHEL7,
+which conflicts with other packages such as `Octave` and causes users troubles
+when installing these packages alongside Mantid.
+
+The package provided for users depends on the system libraries leaving the
+mantid package to just ship the mantid libraries.
+
+In addition, 
+for RHEL 7 (and its variants CentOS, Scientfic Linux) a
+[Software Collection](https://www.softwarecollections.org/en/) is used to provide
+a gcc-7 allowing C++17-enabled builds.
+
+Pros:
+* Easier developer setup.
+* Smaller user packages. The system libraries are shared for all packages resulting
+  in a much smaller package to install.
+* (Mostly) managed for us.
+* Simply commands to install packages (apt/yum)
+
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* Versions become outdated very quickly. Facilities tend to use
+  older distributions that do not have up to date versions of packages.
+* Upgrading/rebuilding package versions can cause conflicts with other packages
+  not related to mantid and even old mantid versions.
+* New dependencies have to be installed manually on developers and builders machines.
+
```

Comment:
- User have to install Mantid repository to pull in deps, they cannot use a .deb/.rpm direct on a new machine